solution/2300-2399/2332.The Latest Time to Catch a Bus/README_EN.md

```diff
@@ -160,6 +160,32 @@ func latestTimeCatchTheBus(buses []int, passengers []int, capacity int) int {
 
 ```
 
+### **JavaScript**
+
+```js
+var latestTimeCatchTheBus = function (buses, passengers, capacity) {
+    buses.sort((a, b) => a - b);
+    passengers.sort((a, b) => a - b);
+    let j = 0;
+    let c;
+    for (const t of buses) {
+        c = capacity;
+        while (c && j < passengers.length && passengers[j] <= t) {
+            c--;
+            j++;
+        }
+    }
+    j--;
+    let latestArrTime = passengers[j];
+    if (c) latestArrTime = buses[buses.length - 1];
+    while (j >= 0 && passengers[j] === latestArrTime) {
+        latestArrTime--;
+        j--;
+    }
+    return latestArrTime;
+};
```

Comment:
I recommend this writing style more because it's easy for users to switch from other languages.

```suggestion
    let ans = c > 0 ? buses[buses.length - 1] : passengers[j];
    while (j >= 0 && ans === passengers[j]) {
        --ans;
        --j;
    }
    return ans;
```

![image](https://github.com/doocs/leetcode/assets/117748716/c191faf8-cdf8-415e-b5ef-54b60cf8c09f)